### PR TITLE
dcrdata/exchanges: use `EndRate` from dcrdex `EpochReportRoute` as dcr rate

### PR DESCRIPTION
**Pull request by ukane-philemon**
Closes https://github.com/decred/dcrdata/issues/2005

### CODE REVIEW

File: exchanges/exchanges.go [hunk 1]
```diff
@@ -2852,7 +2861,7 @@ func (dcr *DecredDEX) setOrderBook(ob *msgjson.OrderBook) {
 
 	dcr.Update(&ExchangeState{
 		BaseState: BaseState{
-			Price: depth.MidGap(),
+			Price: dcr.lastRate,
```
[the remark this answers is not in the document]
Comment (ukane-philemon) [reply exchanges/exchanges.go:2864]: imo we should have a sane value to display before we get the first candle which might or might not take a long time. mid gap is reasonable.

[the remark this answers is not in the document]
Comment (buck54321) [reply exchanges/exchanges.go:2864]: We need to at least skip the update if the orderbook is empty, in which case `MidGap` defaults to `1`, which is not reasonable.

[the remark this answers is not in the document]
Comment (ukane-philemon) [reply exchanges/exchanges.go:2864]: Oh yea.

[the remark this answers is not in the document]
Comment (ukane-philemon) [reply exchanges/exchanges.go:2872]: Fixed in: https://github.com/decred/dcrdata/pull/2007/commits/b4d82d0f72c811d0f04b55a21a76b387e960f948

File: exchanges/exchanges.go [hunk 2]
```diff
@@ -2850,9 +2858,18 @@ func (dcr *DecredDEX) setOrderBook(ob *msgjson.OrderBook) {
 
 	depth := dcr.wsDepthSnapshot()
 
+	if dcr.lastRate == 0 {
+		// Use mid gap as a sane default if the orderbook is not empty.
+		midGap := depth.MidGap()
+		if midGap == 1 {
```
Comment (buck54321) [inline exchanges/exchanges.go:2864]: Probably better to do `if len(ob.Orders) == 0`, but whatever.

**Review by buck54321 — APPROVED** [review]

**Review by buck54321 — APPROVED** [review]

**Pull request merged**